src/components/RenderPipeline/RenderPipelineV2.tsx

```diff
@@ -0,0 +1,74 @@
+import { useStore } from "@/src/types/StoreContext";
+import { MergeNode } from "@/src/components/RenderPipeline/MergeNode";
+import { useRenderTarget, useRenderTargets } from "@/src/hooks/renderTarget";
+import { observer } from "mobx-react-lite";
+import { WebGLRenderTarget } from "three";
+import { BlockStackNode } from "./BlockStackNode";
+import { LayerV2 } from "@/src/types/Layer/LayerV2";
+
+type Props = {
+  renderTargetZ: WebGLRenderTarget;
+};
+
+export const RenderPipelineV2 = observer(function RenderPipeline({
+  renderTargetZ,
+}: Props) {
+  const store = useStore();
+  const layers = store.layers as LayerV2[];
+  if (layers.length === 0) return null;
+  return <LayerNode layer={layers[0]} destinationTarget={renderTargetZ} />;
+});
+
+type LayerNodeProps = {
+  layer: LayerV2;
+  destinationTarget: WebGLRenderTarget;
+};
+
+function LayerNode({ layer, destinationTarget }: LayerNodeProps) {
```

Comment:
You will need to make this component a Mobx observer component in order for it to respond to changes in layer.maxConcurrentBlocks, which is an observable property used in useRenderedTargets. In general, you can make any react component an observer safely.  It will memoize the component at a minimum which is a nice performance benefit, especially here in our rendering pipeline.